CRM/Utils/Recent.php

```diff
@@ -168,11 +233,23 @@ private static function getIcon($type, $subType) {
    * @param \Civi\Core\Event\PostEvent $event
    */
   public static function on_hook_civicrm_post(\Civi\Core\Event\PostEvent $event) {
-    if ($event->action === 'delete' && $event->id && CRM_Core_Session::getLoggedInContactID()) {
-      // Is this an entity that might be in the recent items list?
-      $providersPermitted = Civi::settings()->get('recentItemsProviders') ?: array_keys(self::getProviders());
-      if (in_array($event->entity, $providersPermitted)) {
-        self::del(['id' => $event->id, 'type' => $event->entity]);
+    if ($event->id && CRM_Core_Session::getLoggedInContactID()) {
+      $entityType = self::normalizeEntityType($event->entity);
+      if ($event->action === 'delete') {
+        // Is this an entity that might be in the recent items list?
+        $providersPermitted = Civi::settings()->get('recentItemsProviders') ?: array_keys(self::getProviders());
+        if (in_array($entityType, $providersPermitted)) {
+          self::del(['entity_id' => $event->id, 'entity_type' => $entityType]);
+        }
+      }
+      elseif ($event->action === 'edit') {
+        if (isset($event->object->is_deleted)) {
+          \Civi\Api4\RecentItem::update()
```

Comment:
@colemanw This seems to have caused a regression with webform_civicrm submitting activities/cases via user checksum. Adding FALSE in checkPermissions here fixes that.

Reply:
- I suspect the permissions for that api might be too tight
-

Reply:
I think the permissions are ok as "access CiviCRM', but yea for a checksum user who has no permissions at all... womp womp.